### PR TITLE
Added a setting to ignore unknown presets features from versions we don't support yet

### PR DESCRIPTION
**Pull request by bobbrow**
Addresses #1963

### CODE REVIEW

File: src/presetsController.ts [hunk 1]
```diff
@@ -1016,16 +1022,25 @@ export class PresetsController {
...
+                        logFunc(` >> ${err.dataPath}: ${err.message}`);
+                    }
                 }
+            };
+            if (validationErrorsAreWarnings) {
+                showErrors(log.warning);
+                return presetsFile;
+            } else {
+                showErrors(log.error);
+                log.error(localize('unsupported.presets.disable', 'If desired, unknown properties and macros can be ignored by using the {0} setting.', "'cmake.allowUnsupportedPresetsVersions'"));
```
Comment (elahehrashedi) [inline src/presetsController.ts:1041]: Maybe we can write something easier to understand, I am ok with the current explanation though.
localize('disable.unsupported.presets', 'In order to ignore the unknown properties and macros, modify the {0} setting.', "'cmake.allowUnsupportedPresetsVersions'")

**Review by elahehrashedi — APPROVED** [review]

**Comment by elahehrashedi** [comment]
@bobbrow I made a small change in the description.

**Comment by bobbrow** [comment]
> @bobbrow I made a small change in the description.

That's fine

File: src/presetsController.ts (hunk 1 above)
Comment (elahehrashedi) [reply src/presetsController.ts]: I did it. Thanks

**Pull request merged**